bridge/gitea/config.go

```diff
@@ -0,0 +1,305 @@
+package gitea
+
+import (
+	"context"
+	"fmt"
+	"path"
+	"regexp"
+	"sort"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/MichaelMure/git-bug/bridge/core"
+	"github.com/MichaelMure/git-bug/bridge/core/auth"
+	"github.com/MichaelMure/git-bug/cache"
+	"github.com/MichaelMure/git-bug/commands/input"
+	"github.com/MichaelMure/git-bug/repository"
+)
+
+var (
+	ErrBadProjectURL = errors.New("bad project url")
+)
+
+func (g *Gitea) ValidParams() map[string]interface{} {
+	return map[string]interface{}{
+		"URL":        nil,
+		"Login":      nil,
+		"CredPrefix": nil,
+		"TokenRaw":   nil,
+	}
+}
+
+func (g *Gitea) Configure(repo *cache.RepoCache, params core.BridgeParams, interactive bool) (core.Configuration, error) {
+	var err error
+	var baseURL, owner, project string
+
+	// get project url
+	switch {
+	case params.URL != "":
+		baseURL, owner, project, err = splitURL(params.URL)
+		if err != nil {
+			return nil, err
+		}
+	default:
+		// terminal prompt
+		if !interactive {
+			return nil, fmt.Errorf("Non-interactive-mode is active. Please specify the gitea project URL via the --url option.")
+		}
+		baseURL, owner, project, err = promptURL(repo)
+		if err != nil {
+			return nil, errors.Wrap(err, "url prompt")
+		}
+	}
+
+	var login string
+	var cred auth.Credential
+
+	switch {
+	case params.CredPrefix != "":
+		cred, err = auth.LoadWithPrefix(repo, params.CredPrefix)
+		if err != nil {
+			return nil, err
+		}
+		l, ok := cred.GetMetadata(auth.MetaKeyLogin)
+		if !ok {
+			return nil, fmt.Errorf("credential doesn't have a login")
+		}
+		login = l
+	case params.TokenRaw != "":
+		token := auth.NewToken(target, params.TokenRaw)
+		login, err = getLoginFromToken(baseURL, token)
+		if err != nil {
+			return nil, err
+		}
+		token.SetMetadata(auth.MetaKeyLogin, login)
+		token.SetMetadata(auth.MetaKeyBaseURL, baseURL)
+		cred = token
+	default:
+		if params.Login == "" {
+			if !interactive {
+				return nil, fmt.Errorf("Non-interactive-mode is active. Please specify the login name via the --login option.")
+			}
+			// TODO: validate username
+			login, err = input.Prompt("Gitea login", "login", input.Required)
+		} else {
+			// TODO: validate username
```

Comment:
todo: at least make sure it's not empty, return an error if so

Reply:
Fixed